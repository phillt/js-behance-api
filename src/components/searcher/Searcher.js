import React, {Component} from "react";
import PropTypes from "prop-types";
import {ifFunc} from "project-helpers";

class Searcher extends Component {
    constructor(props) {
        super(props);
        this.state = {
            searchQuery : props.initialValue
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleOnEnter = this.handleOnEnter.bind(this);
    }

    /**
     * Commits changes to the state from the value in the event.
     *
     * @param {object} event - Takes in the event object.
     */
    handleChange(event) {
        const {name, value} = event.target;

        this.setState({[name] : value});
    }

    /**
     * Will call the props function onEnter if the key pressed was the enter key.
     *
     * @param {object} event - The event object.
     */
    handleOnEnter(event) {
        if (event.keyCode === 13){
            // enter key was pressed
            ifFunc(this.props.onEnter, this.state);
        }
    }

    render() {
        return (
            <div>
                <label>
                    Search
                    <input
                        onKeyUp={this.handleOnEnter}
                        onChange={this.handleChange}
                        type={"text"}
                        name={"searchQuery"}
                        value={this.state.searchQuery}
                    />
                </label>
            </div>
        )
    }
}

Searcher.propTypes = {
    onEnter : PropTypes.func, // Handle when user hits enter
    initialValue: PropTypes.string // The initial value
};

export default (Searcher);