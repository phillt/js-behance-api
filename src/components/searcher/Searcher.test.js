import React from "react";
import renderer from "react-test-renderer";
import Searcher from "./searcher.js";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, {shallow} from "enzyme";

Enzyme.configure({ adapter: new Adapter() });

describe("Searcher", ()=>{

    it('Default value is set when initialValue prop is created', () => {
        const searchComponent = renderer.create(<Searcher initialValue={"hello"} />);

        let tree = searchComponent.toJSON();
        expect(tree).toMatchSnapshot();

    });

    it('State value changes onChange', () => {
        const searchComponent = shallow(<Searcher initialValue={""} />);

        // Invoke on change.
        const event = {
            preventDefault() {},
            target: { value: 'hello', name: "searchQuery"}
        };
        searchComponent.find('input').simulate('change', event);

        expect(searchComponent.state().searchQuery).toEqual("hello");

    });

    it('Should call function on enter', ()=>{
        const onEnterMock = jest.fn();
        const searchComponent = shallow(<Searcher initialValue={""} onEnter={onEnterMock} />);
        const event = {
            preventDefault() {},
            target: { value: '', name: "searchQuery"},
            keyCode: 13
        };
        searchComponent.find('input').simulate('keyup', event);
        expect(onEnterMock).toBeCalledWith({"searchQuery": ""});
    })
});


