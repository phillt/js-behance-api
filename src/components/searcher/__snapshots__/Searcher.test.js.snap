// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searcher Default value is set when initialValue prop is created 1`] = `
<div>
  <label>
    Search
    <input
      name="searchQuery"
      onChange={[Function]}
      onKeyUp={[Function]}
      type="text"
      value="hello"
    />
  </label>
</div>
`;
